stix_shifter/stix_translation/src/modules/carbonblack/README.md

```diff
@@ -0,0 +1,377 @@
+['process_name:cmd.exe']
+{'queries': ['process_name:cmd.exe'], 'parsed_stix': [{'attribute': 'process:name', 'comparison_operator': '=', 'value': 'cmd.exe'}]}
+```
+
+Note that because the carbonblack api is synchronous the search id is the same as the translated query.
+```
+$ python3 main.py transmit carbonblack '{"host": "example.carbonblack.io", "port":443}' '{"auth": {"token":"000000000000000000000000000"}}' query "process_name:cmd.exe"
+{'success': True, 'search_id': 'process_name:cmd.exe'}
+```
+
+```
+$ python3 main.py transmit carbonblack '{"host": "example.carbonblack.io", "port":443}' '{"auth": {"token":"0000000000000000000000000000000000000000"}}' results 'process_name:cmd.exe' 0 1
+{
+  "terms": [
+    "process_name:cmd.exe"
+  ],
+  "results": [
+    {
+      "process_md5": "5746bd7e255dd6a8afa06f7c42c1ba41",
+      "sensor_id": 49,
+      "filtering_known_dlls": true,
+      "modload_count": 3,
+      "parent_unique_id": "00000031-0000-09cc-01d4-b1e61979dd7c-000000000001",
+      "emet_count": 0,
+      "alliance_score_srstrust": -100,
+      "cmdline": "C:\\Windows\\system32\\cmd.exe /c tasklist",
+      "alliance_updated_srstrust": "2018-04-05T16:04:34Z",
+      "filemod_count": 0,
+      "id": "00000031-0000-0768-01d4-b1e6197c3edd",
+      "parent_name": "cmd.exe",
+      "parent_md5": "000000000000000000000000000000",
+      "group": "lab1",
+      "parent_id": "00000031-0000-09cc-01d4-b1e61979dd7c",
+      "hostname": "lab1-host1",
+      "last_update": "2019-01-22T00:04:52.937Z",
+      "start": "2019-01-22T00:04:52.875Z",
+      "alliance_link_srstrust": "https://example.com",
+      "comms_ip": 212262914,
+      "regmod_count": 0,
+      "interface_ip": 183439304,
+      "process_pid": 1896,
+      "username": "SYSTEM",
+      "terminated": true,
+      "alliance_data_srstrust": [
+        "5746bd7e255dd6a8afa06f7c42c1ba41"
+      ],
+      "process_name": "cmd.exe",
+      "emet_config": "",
+      "last_server_update": "2019-01-22T00:07:07.064Z",
+      "path": "c:\\windows\\system32\\cmd.exe",
+      "netconn_count": 0,
+      "parent_pid": 2508,
+      "crossproc_count": 2,
+      "segment_id": 1548115627056,
+      "host_type": "workstation",
+      "processblock_count": 0,
+      "os_type": "windows",
+      "childproc_count": 4,
+      "unique_id": "00080031-0000-0748-01d4-b1e61c7c3edd-016872e1cb30"
+    }
+  ],
+
+  "elapsed": 0.05147600173950195,
+  "comprehensive_search": true,
+  "all_segments": true,
+  "total_results": 1,
+  "highlights": [],
+  "facets": {},
+  "tagged_pids": {},
+  "start": 0,
+  "incomplete_results": false,
+  "filtered": {}
+}
+```
+
+Then the results can be translated with the following command: (Note just the json array associated with the results key should be copied into this command)
+
+## Example STIX Output Format
+
+```
+$ python3 main.py translate carbonblack results '{"id": "blah"}' '<copied results from previous query>'
```

Comment:
You may want to indicate that '{"id": "blah"}' should be the identity object that is shown below as the first object in the 'objects' array.